ui/src/main/kotlin/edu/stanford/spezi/ui/AsyncImageResource.kt

```diff
@@ -0,0 +1,49 @@
+package edu.stanford.spezi.ui
+
+import androidx.compose.foundation.Image
+import androidx.compose.foundation.layout.Box
+import androidx.compose.foundation.layout.BoxScope
+import androidx.compose.foundation.layout.fillMaxSize
+import androidx.compose.runtime.Composable
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.layout.ContentScale
+import coil.compose.SubcomposeAsyncImage
+
+/**
+ * Composable function to display an image from a remote url
+ */
+@Composable
+fun AsyncImageResource(
```

Comment:
Okay - not I see it here, that it is in fact a Composable - I would usually expect it to not be one though 😅 similar as to the ImageResource, I'd prefer an approach where a viewModel or even some service could define a resource and we could then specify loading/error ui when we actually define the UI around that AsyncImageResource, right?